contracts/Scroll_SpokePool.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "./SpokePool.sol";
+import "@scroll-tech/contracts/L2/gateways/IL2GatewayRouter.sol";
+
+contract Scroll_SpokePool is SpokePool {
+    IL2GatewayRouter public l2GatewayRouter;
+    uint32 public l1GasLimit;
+
+    /**************************************
+     *               EVENTS               *
+     **************************************/
+
+    event ScrollTokensBridged(address indexed token, address indexed receiver, uint256 amount);
+
+    /**************************************
+     *          PUBLIC FUNCTIONS          *
+     **************************************/
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(
+        address _wrappedNativeTokenAddress,
+        uint32 _depositQuoteTimeBuffer,
+        uint32 _fillDeadlineBuffer
+    ) SpokePool(_wrappedNativeTokenAddress, _depositQuoteTimeBuffer, _fillDeadlineBuffer) {} // solhint-disable-line no-empty-blocks
+
+    /**
+     * @notice Construct the Scroll SpokePool.
+     * @param _l2GatewayRouter Standard bridge contract.
+     * @param _initialDepositId Starting deposit ID. Set to 0 unless this is a re-deployment in order to mitigate
+     * @param _crossDomainAdmin Cross domain admin to set. Can be changed by admin.
+     * @param _hubPool Hub pool address to set. Can be changed by admin.
+     */
+    function initialize(
+        IL2GatewayRouter _l2GatewayRouter,
+        uint32 _initialDepositId,
+        address _crossDomainAdmin,
+        address _hubPool
+    ) public initializer {
+        __SpokePool_init(_initialDepositId, _crossDomainAdmin, _hubPool);
+        l2GatewayRouter = _l2GatewayRouter;
+        l1GasLimit = 250_000;
+    }
+
+    /**************************************
+     *         INTERNAL FUNCTIONS         *
+     **************************************/
+
+    function _bridgeTokensToHubPool(uint256 amountToReturn, address l2TokenAddress) internal virtual override {
+        IL2GatewayRouter _l2GatewayRouter = l2GatewayRouter;
+
+        // The scroll bridge handles arbitrary ERC20 tokens and is mindful of
+        // the official WETH address on-chain. We don't need to do anything specific
+        // to differentiate between WETH and a separate ERC20.
+        // Note: This happens due to the L2GatewayRouter.getERC20Gateway() call
+        _l2GatewayRouter.withdrawERC20{ value: msg.value }(l2TokenAddress, hubPool, amountToReturn, l1GasLimit);
```

Comment:
We have to send ETH on every withdrawal? This is sort of atypical? Are they finalizing these withdrawals for us?

Reply:
My understanding from their documentation is that they are finalizing the withdrawals for us. The docs aren't fully clear, but we are paying a relayer fee with the payable value:

```
    /// @notice Withdraw of some token to a recipient's account on L1.
    /// @dev Make this function payable to send relayer fee in Ether.
    /// @param token The address of token in L2.
    /// @param to The address of recipient's account on L1.
    /// @param amount The amount of token to transfer.
    /// @param gasLimit Unused, but included for potential forward compatibility considerations.
    function withdrawERC20(
        address token,
        address to,
        uint256 amount,
        uint256 gasLimit
    ) external payable;
```

Reply:
@mrice32 I was able to update this such that we can call these contracts without payable. We have a separate script now for finalizing scroll txns and that should be able to be integrated into the finalizer.